assets/src/dashboard/app/reducer/templates.js

```diff
@@ -39,20 +45,51 @@ export const defaultTemplatesState = {
 
 function templateReducer(state, action) {
   switch (action.type) {
-    case ACTION_TYPES.CREATE_TEMPLATE_FROM_STORY: {
-      return state;
-    }
-    case ACTION_TYPES.LOADING_TEMPLATES: {
+    case ACTION_TYPES.LOADING_TEMPLATES:
+    case ACTION_TYPES.CREATING_TEMPLATE_FROM_STORY: {
       return {
         ...state,
         isLoading: action.payload,
       };
     }
+
+    case ACTION_TYPES.CREATE_TEMPLATE_FROM_STORY_SUCCESS: {
+      return {
+        ...state,
+      };
+    }
+
+    case ACTION_TYPES.FETCH_MY_TEMPLATES_FAILURE:
     case ACTION_TYPES.FETCH_TEMPLATES_FAILURE:
+    case ACTION_TYPES.CREATE_TEMPLATE_FROM_STORY_FAILURE:
+      return {
+        ...state,
+        error: action.payload,
+      };
+
+    case ACTION_TYPES.FETCH_MY_TEMPLATES_SUCCESS: {
+      const fetchedTemplatesById = action.payload.savedTemplates.map(
+        ({ id }) => id
+      );
+
+      const combinedTemplateIds =
+        action.payload.page === 1
+          ? fetchedTemplatesById
+          : [...state.savedTemplatesById, ...fetchedTemplatesById];
+
+      const uniqueTemplateIds = [...new Set(combinedTemplateIds)];
```

Comment:
Neat way to unique them.

Reply:
this was Dillon! but yes so much better.